polyphemus/lib/polyphemus.rb

```diff
@@ -17,4 +17,31 @@ def setup_db
     @db.extension :pg_json
     @db.pool.connection_validation_timeout = -1
   end
+
+  def setup_yabeda
+    Yabeda.configure do
+      group :polyphemus do
+        gauge :last_command_completion do
+          comment "Unix time of last time command was completed"
+          tags [:command, :status]
+        end
+      end
+    end
+
+    super
+  end
+
+  def run_command(config, *args, &block)
+    cmd, cmd_args, cmd_kwds = find_command(*args)
+
+    begin
+      super
+      Yabeda.polyphemus.last_command_completion.set({ command: cmd.class.name, status: 'success' }, Time.now.to_i)
```

Comment:
We collect the time of last completion since _sampling_ this is fairly safe.  We can alert when this time value is older than a certain range (no successful runs after 12 hours?) or when the sampled failed status is more recent than the success.

Reply:
Would the alerts be something that are set up in Prometheus? So raw data collection == Yabeda, rules / notifications  == Prometheus, visualization / charting of trends == Grafana?

Reply:
Correct.  Each part is specialized to its specific role, and to be honest, _each part is replacable_.  Grafana can collect data from other data sources, and yabeda can push to other collectors (statsd, for instance, is an alternative).

The nice thing is, this architecture does not lock us into a hard corner in terms of choices.